festim/temperature/temperature_solver.py

```diff
@@ -56,6 +56,17 @@ def __init__(
         self.boundary_conditions = []
         self.sub_expressions = []
 
+    @property
+    def initial_condition(self):
+        return self._initial_condition
+
+    @initial_condition.setter
+    def initial_condition(self, value):
+        if isinstance(value, (int, float, sp.Expr)):
+            self._initial_condition = festim.InitialCondition(field="T", value=value)
+        else:
+            self._initial_condition = value
```

Comment:
If I'm correct, users can provide `F.InitialCondition` with any `field` if they set directly `initial_condition=F.InitialCondition(...)`. Is there any point in restricting the allowable `field` for the initial condition in the heat transfer problem?

Reply:
This is correct. What do you mean by "restricting the allowable field"?

Reply:
I mean, we can add a check for the `field` value. If it's not `T` than raise an error. Though, such restriction doesn't make a lot of sense

Reply:
Since in this specific case `field` isn't even required (users could put anything in there) I don't think that's a good idea.
What we could do is make different classes `InitialTemperature` and `InitialConcentration` (a bit like the approach we hvae in `fenicsx`). For `InitialTemperature` we wouldn't need to specify field.

Maybe worth doing it in another PR? Maybe it's not even needed at all

Reply:
`InitialTemperature` and `InitialConcentration` sound interesting. However, I assume that the current users should be familiar with `InitialCondition` as it is, if they use it. Possibly, these new classes can be considered later, yes